modules/higherKindCore/src/main/scala/tofu/syntax/monadic.scala

```diff
@@ -51,6 +52,10 @@ object monadic extends TupleSemigroupalSyntax with ApplicativeSyntax with MonadS
     def map2[B, Z](fb: F[B])(f: (C, B) => Z)(implicit F: Apply[F]): F[Z]                           = F.map2(fa, fb)(f)
     def map2Eval[B, Z](fb: cats.Eval[F[B]])(f: (C, B) => Z)(implicit F: Apply[F]): cats.Eval[F[Z]] =
       F.map2Eval(fa, fb)(f)
+
+    def replicate_(count: Long)(implicit F: Applicative[F], FD: Defer[F]): F[Unit] =
+      if (count <= 0) unit[F]
+      else productR(FD.defer(replicate_(count - 1)))
```

Comment:
`fa.productR(FD.defer(...))` reads much more easily IMHO even though it is one allocation overhead.

Reply:
No, there is no allocation overhead, still I'm not agree it's more readable